libafl_nyx/src/nyx_bridge.rs

```diff
@@ -1,150 +1,162 @@
-        Ok(x) => {Some(x)}
+    let cfg = config;
+    println!("{}", cfg);
+}
+
+fn nyx_process_start(
+    sharedir_r_str: &str,
+    workdir_r_str: &str,
+    worker_id: u32,
+    cpu_id: u32,
+    create_snapshot: bool,
+    input_buffer_size: Option<u32>,
+    input_buffer_write_protection: bool,
+) -> Option<NyxProcess> {
+    match NyxProcess::process_start(
+        sharedir_r_str,
+        workdir_r_str,
+        worker_id,
+        cpu_id,
+        create_snapshot,
+        input_buffer_size,
+        input_buffer_write_protection,
+    ) {
+        Ok(x) => Some(x),
         Err(msg) => {
             println!("[!] libnyx failed to initialize QEMU-Nyx: {}", msg);
             None
-        },
+        }
     }
 }
 
-
-pub fn nyx_new(sharedir: &str, workdir: &str, cpu_id: u32, input_buffer_size: u32, input_buffer_write_protection: bool) -> Option<NyxProcess> {
-    nyx_process_start(sharedir, workdir, 0, cpu_id, false, Some(input_buffer_size), input_buffer_write_protection)
-}
-
-
-
-pub fn nyx_new_parent(sharedir: &str, workdir: &str, cpu_id: u32, input_buffer_size: u32, input_buffer_write_protection: bool) -> Option<NyxProcess> {
-    nyx_process_start(sharedir, workdir, 0, cpu_id, true, Some(input_buffer_size), input_buffer_write_protection)
-}
-
-
-pub fn nyx_new_child(sharedir: &str, workdir: &str, cpu_id: u32, worker_id: u32) -> Option<NyxProcess> {
+pub fn nyx_new(
+    sharedir: &str,
+    workdir: &str,
+    cpu_id: u32,
+    input_buffer_size: u32,
+    input_buffer_write_protection: bool,
+) -> Option<NyxProcess> {
+    nyx_process_start(
+        sharedir,
+        workdir,
+        0,
+        cpu_id,
+        false,
+        Some(input_buffer_size),
+        input_buffer_write_protection,
+    )
+}
+
+pub fn nyx_new_parent(
+    sharedir: &str,
+    workdir: &str,
+    cpu_id: u32,
+    input_buffer_size: u32,
+    input_buffer_write_protection: bool,
+) -> Option<NyxProcess> {
+    nyx_process_start(
+        sharedir,
+        workdir,
+        0,
+        cpu_id,
+        true,
+        Some(input_buffer_size),
+        input_buffer_write_protection,
+    )
+}
+
+pub fn nyx_new_child(
+    sharedir: &str,
+    workdir: &str,
+    cpu_id: u32,
+    worker_id: u32,
+) -> Option<NyxProcess> {
     if worker_id == 0 {
         println!("[!] libnyx failed -> worker_id=0 cannot be used for child processes");
-        return None
-    }
-    else{
+        return None;
+    } else {
         nyx_process_start(sharedir, workdir, worker_id, cpu_id, true, None, false)
     }
 }
- 
-
-
-pub fn nyx_get_aux_buffer(nyx_process: NyxProcess)-> *mut u8 {
 
-        return nyx_process.aux_buffer_as_mut_ptr();
+pub fn nyx_get_aux_buffer(nyx_process: &NyxProcess) -> *mut u8 {
+    return nyx_process.aux_buffer_as_mut_ptr();
 }
 
-
-pub fn nyx_get_input_buffer(mut nyx_process: NyxProcess) -> *mut u8 {
-
-        return nyx_process.input_buffer_mut().as_mut_ptr();
+pub fn nyx_get_input_buffer(nyx_process: &NyxProcess) -> *mut u8 {
```

Comment:
Why do we want to wrap nyx?

Reply:
to maintain the same set of API with what's in `afl++`'s nyx mode, just make my life easier :)